plover/dictionary/rtfcre_dict.py

```diff
@@ -13,326 +13,134 @@
-            result = '{&' + contents + '}'
-            self._whitespace = prev
-            return result
-            
-        if command == 'cxp':
-            prev = self._whitespace
-            self._whitespace = False
-            contents = self(contents)
-            if contents is None:
-                return None
-            self._whitespace = prev
-            stripped = contents.strip()
-            if stripped in ['.', '!', '?', ',', ';', ':']:
-                return '{' + stripped + '}'
-            if stripped == "'":
-                return "{^'}"
-            if stripped in ['-', '/']:
-                return '{^' + contents + '^}'
-            # Show unknown punctuation as given.
-            return '{^' + contents + '^}'
-        
-        if command == 'cxsvatdictflags' and 'N' in contents:
-            return '{-|}'
-        
-        # unrecognized commands
-        if ignore:
-            return ''
-        else:
-            return self(contents)
-
-    def _re_handle_eclipse_command(self, m):
-        r'({[^\\][^{}]*})'
-        return m.group()
-
-    # caseCATalyst doesn't put punctuation in \cxp so we will treat any 
-    # isolated punctuation at the beginning of the translation as special.
-    def _re_handle_punctuation(self, m):
-        r'^([.?!:;,])(?=\s|$)'
-        if self._whitespace:
-            result = '{%s}' % m.group(1)
-        else:
-            result = m.group(1)
-        return result
-
-    def _re_handle_text(self, m):
-        r'[^{}\\\r\n]+'
-        text = m.group()
-        if self._whitespace:
-            text = self._multiple_whitespace_pattern.sub(r'{^\1^}', text)
+        needed, replacement = self._format_for_lastindex[m.lastindex]
+        return replacement.format(*(self._escape_fn(m.group(g)) for g in needed))
+
+
+class TranslationFormatter:
+
+    TO_ESCAPE = (
+        (r'([\\{}])', r'\\\1'   ),
+        (r'\n\n'    , r'\\par ' ),
+        (r'\n'      , r'\\line '),
+        (r'\t'      , r'\\tab ' ),
+    )
+    ATOMS_FORMATTERS = (
+        # Note order matters!
+        (r'{\.}'                       , r'{{\cxp. }}'                 ),
+        (r'{!}'                        , r'{{\cxp! }}'                 ),
+        (r'{\?}'                       , r'{{\cxp? }}'                 ),
+        (r'{\,}'                       , r'{{\cxp, }}'                 ),
+        (r'{:}'                        , r'{{\cxp: }}'                 ),
+        (r'{;}'                        , r'{{\cxp; }}'                 ),
+        (r'{\^\^?}'                    , r'{{\cxds}}'                  ),
+        (r'{\^([^^}]*)\^}'             , r'{{\cxds {0}\cxds}}'         ),
+        (r'{\^([^^}]*)}'               , r'{{\cxds {0}}}'              ),
+        (r'{([^^}]*)\^}'               , r'{{{0}\cxds}}'               ),
+        (r'{-\|}'                      , r'\cxfc '                     ),
+        (r'{>}'                        , r'\cxfls '                    ),
```

Comment:
`\cxfl`, I don't see a `\cxfls` in the spec.

Reply:
Good catch.